tests/test_input_outputs.py

```diff
@@ -296,13 +296,16 @@ def _fit_keras_model(self, X, y, sample_weight, warm_start, **kwargs):
 
 
 @pytest.mark.parametrize(
-    "X_dtype", ["float32", "float64", "int64", "int32", "uint8", "uint16"]
+    "X_dtype",
+    ["float32", "float64", "int64", "int32", "uint8", "uint16", "object"],
 )
 @pytest.mark.parametrize(
-    "y_dtype", ["float32", "float64", "int64", "int32", "uint8", "uint16"]
+    "y_dtype",
+    ["float32", "float64", "int64", "int32", "uint8", "uint16", "object"],
 )
 @pytest.mark.parametrize(
-    "s_w_dtype", ["float32", "float64", "int64", "int32", "uint8", "uint16"]
+    "s_w_dtype",
+    ["float32", "float64", "int64", "int32", "uint8", "uint16", "object"],
```

Comment:
These tests now take a _long_ time with all of these dtypes. I think we need to split this up into:
1) Test a single dtype at a time for all inputs (i.e. `y`, `X` and `sw` all have `float32`, etc.).
2) Test that mixed dtypes work (we can do 1 float and 1 int for each input, giving 6 tests)

I'm not sure where/how we need to be testing for `run_eagerly`, but if we can reduce that parametrization as well I think that would be nice.

Reply:
Yeah, I've noticed that these tests take a long time too.

I think `run_eagerly` should be tested, certainly with different input types. `run_eagerly` controls if the program is compiled in Tensorflow ahead of time, and if TF can do dynamic type inference.

Reply:
Do you think it needs to be tested against both of the tests proposed above, or can we pick the higher risk one?

Reply:
I think `run_eagerly` only needs to be tested on (2) above. But you could include one set of parameters with the same dtype just make sure it passes:

``` python
@pytest.mark.parametrize("y_dtype", ["float32", "float64", "uint8", "int16"])
@pytest.mark.parametrize("run_eagerly", [True, False])
def test_mixed_dtypes(y_dtype, run_eagerly):
    n, d = 100, 10
    n_classes = 4
    X = np.random.uniform(size=(n, d)).astype("float32")
    y = np.random.choice(n_classes, size=n).astype(y_dtype)
    est = KerasClassifier(..., run_eagerly=run_eagerly)
    est.fit(X, y)
    est.score(X, y)
```

Reply:
I like that. Let's do that next PR.

Reply:
By the way @stsievert, if you want tests to run faster, you can do:
```python3
pip install pytest-xdist
pytest -n auto
```